src/sentry/migrations/0021_hide_environment_none_20191126.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+def hide_environent_none(apps, schema_editor):
+    """
+    Hide environments that are named none, since they're blacklisted and no longer can be created.
+
+    We should iterate over each environment row individually in python instead so that we don't lock the DB up. This is
+    far slower but much safer
+    """
+    Environment = apps.get_model("sentry", "Environment")
+    EnvironmentProject = apps.get_model("sentry", "EnvironmentProject")
+    for env in RangeQuerySetWrapperWithProgressBar(Environment.objects.all()):
+        if env.name == 'none':
+            for project in EnvironmentProject.objects.filter(environment_id=env.id):
```

Comment:
Since we're updating less than 30 rows total, should be fine to not use the ranged wrapper here

Reply:
We should only use the ranged wrapper when there's over 1000 rows right?

Reply:
There's no specific number we aim for, just "a lot". You're right that there's no point at < 1000 since that's the default step.